### ISSUE TITLE
Add e custom hook to allow to add custom data

### ISSUE THREAD

**Issue by dnlup** [immobiliare/fastify-sentry#299]
I think it would be nice to expose an option to pass a custom function that can be used to enrich the event with custom data.

**Comment by dnlup** [comment]
@fernandesGabriel Yes, I imagined the function could be useful to add more info, like the tags you are mentioning. The only difference with the previous version of the plugin would be that you don't need to implement a custom handler.

**Comment by fernandesGabriel** [comment]
That would be perfect @dnlup, I just upgraded to the new version of the plugin, and just noticed that we also have the use case you mention here.

**Comment by dnlup** [comment]
@fernandesGabriel I was thinking, wouldn't the `beforeSend` (the SDK one) option be enough for this use case?

**Comment by noahsark769** [comment]
Chiming in that this would be great to have as well! Either a way to expose beforeSend, or a way to expose adding custom context to errors

**Comment by noahsark769** [comment]
The workaround for now is to use `beforeSend` as mentioned above:

```ts
void app.register(fastifySentryPlugin, {
  dsn: "...",
  ...,
  beforeSend(event, hint) {
    // Can do custom processing based on `hint.originalException`
    return {
      ...event,
      extra: {
        ...event.extra,
        hello: "world"
      },
    };
  },
```

That `hello: "world"` will show up under "Additional Data" in Sentry 👍

**Comment by dnlup** [comment]
Thanks @noahsark769 for the feedback. So maybe something that allows us to use the `scope` api instead of manipulating the raw `event` should be what we are looking for.
